centreon/src/Core/Dashboard/Application/UseCase/FindPerformanceMetricsData/FindPerformanceMetricsData.php

```diff
@@ -0,0 +1,170 @@
+                $performanceMetricsData = $this->findPerformanceMetricsDataAsNonAdmin($request, $accessGroups);
+            }
+            $presenter->presentResponse($this->createResponse($performanceMetricsData));
+        } catch (MetricException $ex) {
+            $this->error('Metric from RRD are not correctly formatted', ['trace' => (string) $ex]);
+            $presenter->presentResponse(new InvalidArgumentResponse($ex->getMessage()));
+        } catch (\Throwable $ex) {
+            $this->error('An error occured while retrieving metrics data', ['trace' => (string) $ex]);
+            $presenter->presentResponse(new ErrorResponse('An error occured while retrieving metrics data'));
+        }
+    }
+
+    /**
+     * find Performance Metrics Data for an admin user.
+     *
+     * @param FindPerformanceMetricsDataRequest $request
+     *
+     * @throws MetricException
+     * @throws \Throwable
+     *
+     * @return PerformanceMetricsData
+     */
+    private function findPerformanceMetricsDataAsAdmin(
+        FindPerformanceMetricsDataRequest $request
+    ): PerformanceMetricsData {
+        $services = $this->metricRepository->findServicesByMetricIds($request->metricIds);
+        $metricsData = [];
+        $this->metricRepositoryLegacy->setContact($this->user);
+        foreach ($services as $service) {
+            /**
+             * array<int<0, max>, array>.
+             */
+            $metricsData[] = $this->metricRepositoryLegacy->findMetricsByService(
+                $service,
+                $request->startDate,
+                $request->endDate
+            );
+        }
+        if (empty($metricsData)) {
+            throw MetricException::metricsNotFound();
+        }
+        $factory = new PerformanceMetricsDataFactory();
+
+        return $factory->createFromRecords($metricsData, $request->metricIds);
+    }
+
+    /**
+     * find Performance Metrics Data for an admin user.
+     *
+     * @param FindPerformanceMetricsDataRequest $request
+     * @param AccessGroup[] $accessGroups
+     *
+     * @throws MetricException
+     * @throws \Throwable
+     *
+     * @return PerformanceMetricsData
+     */
+    private function findPerformanceMetricsDataAsNonAdmin(
+        FindPerformanceMetricsDataRequest $request,
+        array $accessGroups
+    ): PerformanceMetricsData {
+        $services = $this->metricRepository->findServicesByMetricIdsAndAccessGroups($request->metricIds, $accessGroups);
+        $metricsData = [];
+        $this->metricRepositoryLegacy->setContact($this->user);
+        foreach ($services as $service) {
+            /**
+             * array<int<0, max>, array>.
+             */
+            $metricsData[] = $this->metricRepositoryLegacy->findMetricsByService(
+                $service,
+                $request->startDate,
+                $request->endDate
+            );
+        }
+        if (empty($metricsData)) {
```

Comment:
This an array, you can be stricter.
```suggestion
        if ([] === $metricsData) {
```